src/dialog/Select.tsx

```diff
@@ -0,0 +1,243 @@
+/** @jsx JSXSlack.h */
+import { Dialog, SelectOption } from '@slack/types'
+import {
+  Optgroup,
+  OptgroupInternal,
+  Option,
+  OptionInternal,
+} from '../block-kit/interactive/Select'
+import { JSXSlack } from '../jsx'
+import { ObjectOutput } from '../utils'
+import { validateElement } from './Dialog'
+import { DialogValidationError } from './error'
+
+interface SelectFragmentProps {
+  children: JSXSlack.Children<any>
+}
+
+interface SelectPropsBase {
+  label: string
+  name: string
+  placeholder?: string
+  required?: boolean
+}
+
+interface SelectProps extends SelectPropsBase {
+  children: JSXSlack.Children<OptionInternal | OptgroupInternal>
+  value?: string
+}
+
+interface ExternalSelectProps extends SelectPropsBase {
+  children?: undefined
+  initialOption?: JSXSlack.Node<OptionInternal> | SelectOption
+  minQueryLength?: number
+}
+
+interface UsersSelectProps extends SelectPropsBase {
+  initialUser?: string
+  children?: undefined
+}
+
+interface ConversationsSelectProps extends SelectPropsBase {
+  initialConversation?: string
+  children?: undefined
+}
+
+interface ChannelsSelectProps extends SelectPropsBase {
+  initialChannel?: string
+  children?: undefined
+}
+
+type DialogElement = Dialog['elements'][0]
+
+type SelectBase = Pick<
```

Comment:
[Slack does not explain about `hint` attribute in JSON attributes for `select` type.](https://api.slack.com/dialogs#attributes_select_elements) But in fact, `hint` seems to be able to use in select type too.

<img width="236" alt="hint" src="https://user-images.githubusercontent.com/3993388/62362568-36a45680-b558-11e9-80a5-e37f23b6fa52.png">